progetto/ClientProject/src/client/Client.java

```diff
@@ -111,67 +156,94 @@ public void AnonymousRegister(){
-                System.err.println(response.getCodice() + response.getMessaggioInfo());
-                break;
-
-            default:
-                //TODO add something here
+        this.broker=broker;
+        try {
+            Registry r = LocateRegistry.getRegistry(this.broker, this.port);
+            this.server_stub = (ServerInterface) r.lookup("ServerInterface");
+            ResponseCode response = server_stub.connect(this.skeleton, myPublicKey);
+
+            switch (response.getCodice()) {
+                case R210:
+                    this.brokerPublicKey = response.getMessaggioInfo();
+                    return true;
+                default:
+                    System.out.println(response.getCodice()+":"+response.getMessaggioInfo()+"  FROM:"+response.getClasseGeneratrice());
+                    return false;
+            }
+        }catch (RemoteException |NotBoundException exc){
+            System.err.println(exc.getClass().getSimpleName());
+            exc.printStackTrace();
+            return false;
         }
     }
 
 
-    public void Subscribe(Topic topic)
+    public void subscribe(Topic topic)
     {
-    }
 
 
-    // *************************************************************************************************************
-    //REMOTE METHOD
 
-    @Override
-    public ResponseCode notify(Message m) /*throws RemoteException*/ {
-        ResponseCode rc;
-        if(m==null) {
-             rc=new ResponseCode(ResponseCode.Codici.R500, ResponseCode.TipoClasse.CLIENT,
-                    "(-) NOT OK Il server ha ricevuto un messaggio vuoto");
-            return rc;
-        }
-         rc=new ResponseCode(ResponseCode.Codici.R200, ResponseCode.TipoClasse.CLIENT,
-                "(+) OK il server ha ricevuto il messaggio");
-        return rc;
+
     }
 
-    /* is throw remoteException necessary?*/
-    @Override
-    public boolean isAlive()/* throws RemoteException*/ {
-        return true;
+    public boolean disconnect(){
+        try {
+            boolean uscita=false;
+            ResponseCode response=server_stub.disconnect(cookie);
+            switch(response.getCodice())
+            {
+                case R200:
+                    uscita=true;
+                    System.out.println(response.getCodice()+":"+response.getClasseGeneratrice()+":"+response.getMessaggioInfo());
+                    break;
+                default:
+                    System.err.println(response.getCodice()+":"+response.getClasseGeneratrice()+":"+response.getMessaggioInfo());
+                    uscita=false;
+                    break;
+            }
+            this.broker=null;
+            return uscita;
+
+        }catch(RemoteException exc){
+            System.err.println(exc.getClass().getSimpleName());
+            return false;
+        }
+
     }
 
-    // *************************************************************************************************************
-    //PRIVATE METHOD
+
```

Comment:
Il metodo retrieveAccount server al client per recuperare il suo account se si è già registrato in precedenza sul broker

Reply:
Per recuperare cosa ?